comp/api/api/apiimpl/internal/agent/agent.go

```diff
@@ -547,11 +548,24 @@ func getDiagnose(w http.ResponseWriter, r *http.Request, diagnoseDeps diagnose.S
 	_ = conn.SetDeadline(time.Time{})
 
 	// Indicate that we are already running in Agent process (and flip RunLocal)
-	diagCfg.RunningInAgentProcess = true
 	diagCfg.RunLocal = true
 
+	var diagnoses []diagnosis.Diagnoses
+	var err error
+
 	// Get diagnoses via API
-	diagnoses, err := diagnose.Run(diagCfg, diagnoseDeps)
+	// TODO: Once API component will be refactored, clean these dependencies
+	collector, ok := diagnoseDeps.Collector.Get()
+	if ok {
+		diagnoses, err = diagnose.RunInAgentProcess(diagCfg, diagnose.NewSuitesDepsInAgentProcess(collector))
+	} else {
+		ac, ok := diagnoseDeps.AC.Get()
+		if ok {
+			diagnoses, err = diagnose.RunInCLIProcess(diagCfg, diagnose.NewSuitesDepsInCLIProcess(diagnoseDeps.SenderManager, diagnoseDeps.SecretResolver, diagnoseDeps.WMeta, ac))
+		} else {
+			err = errors.New("collector or autoDiscovery not found")
+		}
+	}
```

Comment:
This code or something like this was running inside diagnose package and controlled by configuration. Do you think it worth to pull it out here? Or you are thinking of changing it later?

Reply:
Ah, I think because of difference dependencies, damn

Reply:
There is an on going work to update the API Component. Collector is available only when running the Agent but not when running JMX fetch. When the work on the API component will be done, I will try to see if I can remove the condition `if ok `. Does it make sense?